examples/KelvinHelmholtz/include/simulation_defines/param/particleConfig.param

```diff
@@ -22,84 +22,123 @@
 
 #pragma once
 
-#include "particles/shapes.hpp"
+#include "particles/startPosition/functors.def"
+#include "particles/manipulators/manipulators.def"
+#include "nvidia/functors/Add.hpp"
+#include "nvidia/functors/Assign.hpp"
 
 namespace picongpu
 {
 
-/** Initial Temperature of the electron plasma in keV
- *  unit: keV */
-const double ELECTRON_TEMPERATURE_keV = 0.0005;
-/** Initial Temperature of the ion plasma in keV
- *  unit keV */
-const double ION_TEMPERATURE_keV = 0.0;
-
-/** Initial particle drift velocity for electrons and ions
- *  Examples:
- *    - No drift is equal to 1.0
- *  unit: none */
-const double PARTICLE_INIT_DRIFT_GAMMA = 1.021;
-const double PARTICLE_INIT_DRIFT_GAMMA_MIDDLE = PARTICLE_INIT_DRIFT_GAMMA;
-
-/* add negativ value to to have two streams with different direction
- * add positiv direction to have two streams in the same direction
- */
-const double PARTICLE_INIT_DRIFT_DIRECTION_MIDDLE = -1.0;
-
-/*change density of middle area in proportion to default density*/
-const double PARTICLE_INIT_DENSITY_FACTOR_MIDDLE = 1.0;
-
-/** Normal vector of the initial drift
- *  length: gets normalized afterwards
- *  unit: none */
-const float_X PARTICLE_INIT_DRIFT_DIRECTION_X = 1.0;
-const float_X PARTICLE_INIT_DRIFT_DIRECTION_Y = 0.0;
-const float_X PARTICLE_INIT_DRIFT_DIRECTION_Z = 0.0;
-
-/** You can manipulate the flow vector of ions by this factor
- *  unit: none */
-const float_X PARTICLE_INIT_DRIFT_IONS = 1.0f;
-
-/** Relative size in the middle which gets the inverse drift
- *  Examples:
- *    - All particles in the simulation drift in one direction: 0.0 or 1.0
- *    - Spacial 1/4 "up", 1/2 "down", 1/4 "up: 0.5
- *  unit: none */
-const double MIDDLE_DRIFT_INV_SIZE_REL = 0.5;
+namespace particles
+{
 
-// Macro Particle Configuration -------------------------------------------
+    /** a particle with a weighting below MIN_WEIGHTING will not
+     *      be created / will be deleted
+     *  unit: none */
+    const float_X MIN_WEIGHTING = 10.0;
 
-/** a particle with a weighting below MIN_WEIGHTING will not
- *      be created / will be deleted
- *  unit: none */
-const float_X MIN_WEIGHTING = 10.0;
+    const uint32_t TYPICAL_PARTICLES_PER_CELL = 25;
 
-namespace particleInitRandomPos
+namespace manipulators
 {
-    /** Count of particles per cell at initial state
-     *  unit: none */
-    const uint32_t NUM_PARTICLES_PER_CELL = 25;
-}
 
-namespace particleInitQuietStart
+    CONST_VECTOR(float_X,3,DriftParamPositive_direction,1.0,0.0,0.0);
```

Comment:
To the last comment: This is one of my next tasks after this pull request but I can definitive not provide this the next weeks.
I will open an issue for this in the next days which describe a little bit more the `free formula` concept (mixing runtime/compiletime functors) we talked about.